### ISSUE TITLE
[UnCLIPImageVariationPipeline] Possible Mismatch in Implementation vs Paper

### ISSUE THREAD

**Issue by Abhinay1997** [huggingface/diffusers#3840]
Hi Team,

There is a possible mismatch in the implementation of [UnCLIPImageVariationPipeline](UnCLIPImageVariationPipeline) w.r.t the Dalle-2 [paper](https://cdn.openai.com/papers/dall-e-2.pdf)

### Context:
Any image can be represented by a bipartite representation $(z_i, X_T)$ where $z_i$ is the CLIP Image embedding of the image and $X_T$ is the noise latent obtained from DDIM Inversion.

So as long as the process is known ( deterministic a.k.a DDIM, $eta=0$), we can reproduce the original image from just $(z_i, X_T) $

From above, it follows that by making the diffusion process random i.e $0 < eta <= 1$ we can generate variations of an image with the same $(z_i, X_T)$

### Possible Mismatch
We need to have $(z_i, X_T)$ to do the above. 

$z_i$ -> obtained with `_encode_image` in `UnCLIPImageVariationPipeline`
$X_T$ -> should be obtained via DDIM Inversion of the UnCLIP decoder but isn't defined in the current implementation.

### How it works currently
Instead of keeping $X_T$ constant, we use `prepare_latents` which returns a random tensor. We can make semantically meaningful image variations because the CLIP Image embeddings are still constant.

## TL;DR

Expected:

Input Image -> [ Image Embeds, Initial Noise Latent ] -> [DECODER, eta > 0 ] -> Variation of input image

Current:

Input Image -> [Image Embeds, Random noise ] -> [DECODER, eta = 1 ] -> Variation of input image.

Please correct me if I'm wrong.

cc: @sayakpaul, @patrickvonplaten

**Comment by patrickvonplaten** [comment]
cc @williamberman here 

Overall we didn't implement `UnCLIPImageVariationPipeline` as noted in the Dalle2 paper, but we followed the official implementation here: https://github.com/Stability-AI/stablediffusion/blob/main/scripts/streamlit/stableunclip.py
